metaStoreApp/app/controllers/api/v1/metadatas_controller.rb

```diff
@@ -1,34 +1,34 @@
 module Api
-    module V1
-        require 'csv'
-        class MetadatasController < MediasController
+  module V1
+    require 'csv'
 
-            def create
-                
-                print "entered\n"
-                Metadata.delete_all
-                saved=[]
-                unsaved=[]
-                CSV.foreach(params[:id], headers: true) do |row|
-                    #add_data row.to_hash
-                    #metadata = Metadata.new(add_data row.to_hash)
-                    #metadata.save
-                    print "\n************************\n"
-                    print row.to_hash
-                    print "\n************************\n"
-                    metadata = Metadata.new(row.to_hash)
-                    if metadata.save
-                        saved.append(row.to_hash)   
-                    else
-                        unsaved.append(row.to_hash)  
-                    end
-                end
-                render json: {status: 'SUCCESS', message:'Loaded metadata',saved_data:saved,unsaved_data:unsaved}, status: :ok 
-            
+    class MetadatasController < ApplicationController
+      def create          
+        Metadata.delete_all
+        saved = []
+        unsaved = []
+        CSV.foreach(params[:id], headers: true) do |row|
+          hash_value = row.to_hash
+          if Media.exists? asset_id: hash_value["media_id"]
+            type = Media.where(asset_id: hash_value["media_id"]).pluck(:media_type)[0]
+            if type == "audio"
+              audio_instance = AudiosController.new
```

Comment:
This is an anti pattern in while writing rails, we should not write code to initialize controllers.